Scripts/libs/game_improvements/interactable.lua

```diff
@@ -4,28 +4,21 @@ __InteractableImprovements_Loaded = true
 -- server: 
 
 local values = {} -- <<not directly accessible for other scripts
-function sm.interactable.setValue(interactable, value)  
-    local currenttick = sm.game.getCurrentTick()
-    values[interactable.id] = {
-        {tick = currenttick, value = {value}}, 
-        values[interactable.id] and (    
-            values[interactable.id][1] ~= nil and 
-            (values[interactable.id][1].tick < currenttick) and 
-            values[interactable.id][1].value or 
-			values[interactable.id][2]
-        ) 
-        or nil
-    }
+function sm.interactable.setValue(interactable, value)
+    values[interactable.id] = values[interactable.id] or {}
+    values[interactable.id].next = value
+    values[interactable.id].changedTick = sm.game.getCurrentTick()
```

Comment:
This is a lot more readable than the previous version. Could you reduce the amount of lookups done here? This function is used everywhere, so making it as fast as possible would be ideal. Perhaps `interactable.publicData` can be used. That way we also don't have a memory leak due to storing values for parts that have been destroyed.

Reply:
Fixed